teammapper-backend/src/map/services/maps.service.ts

```diff
@@ -101,21 +98,21 @@ export class MapsService {
   ): Promise<MmpNode[]> {
     if (!mapId || nodes.length === 0) return Promise.reject()
 
-      const reducer = async (previousPromise: Promise<MmpNode[]>, node: MmpNode): Promise<MmpNode[]> => {
-        const accCreatedNodes = await previousPromise;
-        if (await this.validatesNodeParentForNode(mapId, node)) {
-          try {
-            const newNode = await this.addNode(mapId, node);
-            return accCreatedNodes.concat([newNode]);
-          } catch (error) {
-            this.logger.warn(`Failed to add node ${node.id} to map ${mapId}: ${error}`);
-            return accCreatedNodes;
-          }
+    const reducer = async (previousPromise: Promise<MmpNode[]>, node: MmpNode): Promise<MmpNode[]> => {
+      const accCreatedNodes = await previousPromise;
+      if (await this.validatesNodeParentForNode(mapId, node)) {
```

Comment:
as we are already catching errors below when the parent is missing, might be ok to remove that here. However, maybe make the catch explicit for certain errors. E.g. if instance of X then catch, otherwise let the error bubble up.

Reply:
I don't think we're explicitly checking if the parent node exists, and I'm not sure upsert would throw an error as the parent node can be undefined, or am I missing something?

Reply:
I am more referring to a wrong parent Id that does not exist. I think upsert will fail here, maybe do a quick test ?